File: Sorts/js/bubble_sort.js
function bubble_sort(arr) {
  for (let i = arr.length - 1; i >= 0; i--) {
    let isWapped = false;
    for (let j = 0; j < i; j++) {
      if (arr[j] > arr[j + 1]) {
        swap(arr, j, j + 1);
        isWapped = true;
      }
    }
    if (isWapped === false) {
      return arr;
    }
  }
  return arr;
}

function swap(arr, indexOne, indexTwo) {
  let temp = arr[indexOne];
  arr[indexOne] = arr[indexTwo];
  arr[indexTwo] = temp;
}
console.log(bubble_sort([4, 6, 1, 15, 13, 11, 9, 7]));

File: Sorts/python/bubble_sort.py
def bubble_sort(arr):
    # iterate through array foward
    for i in range(len(arr)):
        iswapped = False
        # iterate though array forward till
        for j in range(len(arr) - i - 1):
            # do a check
            if arr[j] > arr[j+1]:
                swapper = arr[j]
                arr[j] = arr[j+1]
                arr[j+1] = swapper
                iswapped = True 
        # check if there no swapper, which every element is sort
        if not iswapped:
            return arr
    return arr

print(bubble_sort([5,3,9,11,2,1]))
